app/V1Module/presenters/UploadedFilesPresenter.php

```diff
@@ -205,10 +255,222 @@ public function actionUpload()
         } catch (Exception $e) {
             $this->uploadedFiles->remove($uploadedFile);
             $this->uploadedFiles->flush();
-            throw new InternalServerException("Cannot move uploaded file to internal server storage");
+            throw new InternalServerException(
+                "Cannot move uploaded file to internal server storage",
+                FrontendErrorMappings::E500_000__INTERNAL_SERVER_ERROR,
+                null,
+                $e
+            );
+        }
+
+        $this->uploadedFiles->flush();
+        $this->sendSuccessResponse($uploadedFile);
+    }
+
+    public function checkStartPartial()
+    {
+        // actually, this is same as upload
+        if (!$this->uploadedFileAcl->canUpload()) {
+            throw new ForbiddenRequestException();
+        }
+    }
+
+    /**
+     * Start new upload per-partes. This process expects the file is uploaded as a sequence of PUT requests,
+     * each one carrying a chunk of data. Once all the chunks are in place, the complete request assembles
+     * them together in one file and transforms UploadPartialFile into UploadFile entity.
+     * @POST
+     * @Param(type="post", name="name", required=true, validation="string:1..255",
+     *        description="Name of the uploaded file.")
+     * @Param(type="post", name="size", required=true, validation="numericint", description="Total size in bytes.")
+     */
+    public function actionStartPartial()
+    {
+        $user = $this->getCurrentUser();
+        $name = $this->getRequest()->getPost("name");
+        $size = (int)$this->getRequest()->getPost("size");
+
+        if (!Strings::match($name, self::FILENAME_PATTERN)) {
+            throw new CannotReceiveUploadedFileException(
+                "File name '$name' contains invalid characters",
+                IResponse::S400_BAD_REQUEST,
+                FrontendErrorMappings::E400_003__UPLOADED_FILE_INVALID_CHARACTERS,
+                ["filename" => $name, "pattern" => self::FILENAME_PATTERN]
+            );
+        }
+
+        $maxSize = 1024 * 1024 * 1024;
+        if ($size < 0 || $size > $maxSize) {
```

Comment:
Separate condition for file-too-big would be nice

Reply:
Actually, I am planning the `file-too-big` error will be introduced with quotas (which are inevitable). At this moment 1G is like "completely off".